apps/settings/js/carrier.js

```diff
@@ -119,6 +119,15 @@ var CarrierSettings = (function() {
       });
     }
 
+    var isThereNoSimCard = Array.prototype.every.call(
+      navigator.mozMobileConnections,
+      function(connection) {
+        return !connection.iccId;
+      }
+    );
+    if (isThereNoSimCard) {
+      dataToggle.disabled = true;
```

Comment:
Should we turn the toggle off before disabling it? It looks weird to me if the toggle is on but disabled.